### PR TITLE
[Merged by Bors] - feat: `negOnePow n = (-1 : ℤ) ^ n`

### PR DESCRIPTION
**Pull request by YaelDillies**
and rename `Int.coe_negOnePow` to `Int.cast_negOnePow` for disambiguation

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Ring/NegOnePow.lean [hunk 1]
```diff
@@ -101,11 +101,17 @@ lemma negOnePow_eq_iff (n₁ n₂ : ℤ) :
...
+  · simp [zpow_mul, zpow_ofNat]
   · rw [zpow_add_one₀ (by norm_num), zpow_mul, zpow_ofNat]
     simp
 
+@[deprecated (since := "2024-10-20")] alias coe_negOnePow := cast_negOnePow
+
+lemma cast_negOnePow_natCast (R : Type*) (n : ℕ) [Ring R] : negOnePow n = (-1 : R) ^ n := by
+  obtain ⟨k, rfl | rfl⟩ := Nat.even_or_odd' n <;> simp [pow_succ, pow_mul]
+
+lemma coe_negOnepow_natCast (n : ℕ) : negOnePow n = (-1 : ℤ) ^ n := cast_negOnePow_natCast ..
```
Comment (b-mehta) [inline Mathlib/Algebra/Ring/NegOnePow.lean:115]:
```suggestion
lemma coe_negOnePow_natCast (n : ℕ) : negOnePow n = (-1 : ℤ) ^ n := cast_negOnePow_natCast ..
```

File: Mathlib/Algebra/Ring/NegOnePow.lean [hunk 2]
```diff
@@ -101,11 +101,17 @@ lemma negOnePow_eq_iff (n₁ n₂ : ℤ) :
...
   rcases even_or_odd' n with ⟨k, rfl | rfl⟩
-  · rw [zpow_mul, zpow_ofNat]
-    simp
+  · simp [zpow_mul, zpow_ofNat]
   · rw [zpow_add_one₀ (by norm_num), zpow_mul, zpow_ofNat]
     simp
 
+@[deprecated (since := "2024-10-20")] alias coe_negOnePow := cast_negOnePow
+
+lemma cast_negOnePow_natCast (R : Type*) (n : ℕ) [Ring R] : negOnePow n = (-1 : R) ^ n := by
```
Comment (b-mehta) [inline Mathlib/Algebra/Ring/NegOnePow.lean:112]:
```suggestion
lemma cast_negOnePow_natCast (R : Type*) [Ring R] (n : ℕ) : negOnePow n = (-1 : R) ^ n := by
```

**Review by b-mehta — APPROVED** [review]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**